### PR TITLE
Add Grafana Oncall support

### PR DESCRIPTION
**Pull request by Konstantinov-Innokentii**

### CODE REVIEW

File: grafana/provider.go
Comment (julienduchesne) [inline grafana/provider.go]: oops, probably not `https://synthetic-monitoring-api.grafana.net`

File: grafana/provider.go
Comment (julienduchesne) [inline grafana/provider.go]: Description is copied as well

File: grafana/provider.go
Comment (julienduchesne) [inline grafana/provider.go]: I'd call that `amixrAPI` even if that's a bit more verbose. I can read the others, but aapi is a bit harder

File: grafana/provider.go
Comment (julienduchesne) [inline grafana/provider.go]: Same here
```suggestion
func createAmixrClient(d *schema.ResourceData) (*aapi.Client, error) {
```

File: grafana/resource_amixr_integration.go
Comment (julienduchesne) [inline grafana/resource_amixr_integration.go]: I think these should be prefixed as well, to be more explicit when autocompleting

```suggestion
func resourceAmixrIntegrationCreate(d *schema.ResourceData, m interface{}) error {
```

File: grafana/provider.go
Comment (julienduchesne) [inline grafana/provider.go]: Oops, removed a cloud resource there

**Review by julienduchesne — COMMENTED** [review]
This is coming along nicely!

File: grafana/provider.go
Comment (inkel) [reply grafana/provider.go]: I would also suggest `ocapi` or `irmapi` instead.

File: grafana/adapters.go
Comment (inkel) [inline grafana/adapters.go]: There's no need for this `if`: calling `range` on a `nil` slice is the same as calling it over an empty slice. What I suggest doing instead is allocating `ret` with the required size:
```suggestion
	ret := make([][]string, 0, len(listSet))
```

File: grafana/data_source_amixr_action.go
Comment (inkel) [inline grafana/data_source_amixr_action.go]: I believe this is going to the wrong page, maybe this is the right URL?
```suggestion
* [HTTP API](https://grafana.com/docs/grafana-cloud/oncall/oncall-api-reference/)
```

File: grafana/provider.go
Comment (julienduchesne) [inline grafana/provider.go]:
```suggestion
	return amixrAPI.New(base_url, aToken)
}
```
You can return immediately, the client will be ignored if there's an err

File: grafana/resource_amixr_route.go
Comment (julienduchesne) [inline grafana/resource_amixr_route.go]: I have an idea to remove all of these blocks, but it's just sugar. We can wait until we migrate all of this to `...Context` functions

**Review by julienduchesne — APPROVED** [review]
I don't see any blockers to merge this. I think we can ship this and go incrementally from there!

File: grafana/data_source_amixr_escalation_chain.go
Comment (inkel) [inline grafana/data_source_amixr_escalation_chain.go]: No need to create a new variable here:

```suggestion
		return errors.New("amixr api client is not configured")
```

File: grafana/data_source_amixr_escalation_chain.go
Comment (inkel) [inline grafana/data_source_amixr_escalation_chain.go]: Nit: error checking should be immediately after the action that might trigger an error is.
```suggestion
	escalationChainsResponse, _, err := client.EscalationChains.ListEscalationChains(options)
	if err != nil {
		return err
	}
```

File: grafana/data_source_amixr_schedule.go
Comment (inkel) [inline grafana/data_source_amixr_schedule.go]:
```suggestion
		return errors.New("amixr api client is not configured")
```

File: grafana/data_source_amixr_schedule.go
Comment (inkel) [inline grafana/data_source_amixr_schedule.go]:
```suggestion
	schedulesResponse, _, err := client.Schedules.ListSchedules(options)
	if err != nil {
		return err
	}
```

File: grafana/data_source_amixr_schedule.go
Comment (inkel) [reply grafana/data_source_amixr_schedule.go]: I've found the same in other files, can you please check those as well?

**Review by inkel — CHANGES_REQUESTED** [review]
I've just left a couple suggestions regarding styling. Also, can you format the example files? I've just checked and the following aren't properly formatted.

```
$ terraform fmt -check -recursive examples
examples/provider/provider-amixr.tf
examples/resources/grafana_amixr_escalation/resource.tf
examples/resources/grafana_amixr_on_call_shift/grafana_amixr_on_call_shift.tf
examples/resources/grafana_amixr_route/resource.tf
examples/resources/grafana_amixr_schedule/resource.tf
```

File: grafana/data_source_amixr_action.go
Comment (Konstantinov-Innokentii) [reply grafana/data_source_amixr_action.go]: The link is correct, action and outgoing webhook are the same

File: grafana/provider_test.go
Comment (inkel) [inline grafana/provider_test.go:88]: I'm curious: where does this `\n` between `must` and `be` come from? Checking the [SDK code](https://github.com/hashicorp/terraform-plugin-sdk/blob/4681738a561387fb0b3aaa69aeb42231383634a0/helper/schema/schema.go#L1874) that raises this error doesn't seem to have a newline, maybe it's the `resource.UnitTest`? 🤔

**Review by inkel — APPROVED** [review]
Excellent work, @Konstantinov-Innokentii!!! Thanks for your patience with all the change requests ❤️ 

Let's wait for @julienduchesne for a final approval.

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**
